challenge_5/C/karanchawla/src/challenge5.c

```diff
@@ -0,0 +1,109 @@
+/*
+Author: Karan Chawla
+Challenege #5
+Expected Running time O(k + n log n)
+where k is the position of the char in the new sorted string
+*/
+
+#include <stdio.h>
+#include <stdlib.h> 
+#include <time.h> 
+#include <string.h>
+
+
+//utility function used by qsort to sort the strings
+int comp (const void * elem1, const void * elem2) 
+{
+    char f = *((char*)elem1);
+    char s = *((char*)elem2);
+    if (f > s) return  1;
+    if (f < s) return -1;
+    return 0;
+}
+
+//function to shuffle the string
+void shuffle(char *array, size_t n)
+{
+    if (n > 1) 
+    {
+        size_t i;
+        for (i = 0; i < n - 1; i++) 
+        {
+            size_t j = i + rand() / (RAND_MAX / (n - i) + 1);
```

Comment:
You're correct `i < n` instead of `n-1` and I can express `j` as `size_t j = rand()%n ;` too. It's just to get a random index <= size of the string.